mcs/class/monodoc/Test/Monodoc.Ecma/EcmaUrlTests.cs

```diff
@@ -454,38 +470,293 @@ public void ExplicitIndexerImplementation ()
 			AssertUrlDesc (ast, "P:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item(System.Int32)");
 		}
 
-		/*		[Test]
-		public void TreeParsabilityTest ()
+		[Test]
+		public void ToEcmaCref_Namespace ()
 		{
-			var rootTree = RootTree.LoadTree ("/home/jeremie/monodoc/");
-			Node result;
-			var generator = new CheckGenerator ();
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Namespace,
+				Namespace = "System.IO"
+			};
 
-			foreach (var leaf in GetLeaves (rootTree.RootNode).Where (IsEcmaNode))
-				AssertUrl (leaf.PublicUrl);
+			AssertEcmaString ("N:System.IO", actual);
 		}
 
-		IEnumerable<Node> GetLeaves (Node node)
+		[Test]
+		public void ToEcmaCref_SimpleType ()
 		{
-			if (node == null)
-				yield break;
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Type,
+				Namespace = "System.IO",
+				TypeName = "Path"
+			};
 
-			if (node.IsLeaf)
-				yield return node;
-			else {
-				foreach (var child in node.Nodes) {
-					if (!string.IsNullOrEmpty (child.Element) && !child.Element.StartsWith ("root:/"))
-						yield return child;
-					foreach (var childLeaf in GetLeaves (child))
-						yield return childLeaf;
-				}
-			}
+			AssertEcmaString ("T:System.IO.Path", actual);
+		}
+
+		[Test]
+		public void ToEcmaCref_NestedType ()
+		{
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Type,
+				Namespace = "System.IO",
+				TypeName = "Path",
+				NestedType = new EcmaDesc { DescKind = EcmaDesc.Kind.Type, TypeName = "TheNestedType" },
```

Comment:
You should also test *multiple nested types*, e.g. [`Mono.DocTest.NestedClass.Double.Triple.Quadruple](https://github.com/mono/mono/blob/master/mcs/tools/mdoc/Test/DocTest-v1.cs#L173), which is insanely nested. :-)